devel/gen_all_horner_cpp_header.m

```diff
@@ -0,0 +1,60 @@
+% Script to make C++ header for looped Horner eval of kernels of all widths,
+% for a particular opts.upsampfac (user to hand-choose below).
+% C++ version (now writes .h), uses constexpr to switch by width w.
+
+% Barnett 4/23/18; now calling Ludvig's loop version from 4/25/18.
+% version including low upsampfac, 6/17/18.
+% Ludvig put in w=4n padding, 1/31/20. Mystery about why d was bigger 2/6/20.
+% C++ header using constexpr of Barbone, replacing *_C_code.m. Barnett 7/16/24.
+clear
+opts = struct();
+
+ws = 2:16;           % list of widths (the driver, rather than toloerances)
+upsampfac = 1.25;    % sigma (upsampling): either 2 (default) or low (eg 5/4)
+
+if upsampfac==2
+  fid = fopen('../src/ker_horner_allw_loop_constexpr.h','w');
+  get_nc_code = 'w + 2 + (w <= 8)';   % must be C++ expression for nc=d+1, not d
+elseif upsampfac==1.25
+  fid = fopen('../src/ker_lowupsampfac_horner_allw_loop_constexpr.h','w');
+  get_nc_code = 'std::ceil(0.6*w + 3.2)';  % must be C++ code for nc=d+1, not d
```

Comment:
`std::ceil`  is `constexpr` and can be evaluated in compile time with C++23 standard.  For C++ standard less than C++23, the generated `nc125` function may not compile. Currently makefile and cmake use C++17.
In case we need to use < C++23, the `nc125` function may need to be something like,
```C++
template<uint8_t w> constexpr auto nc125() noexcept { constexpr uint8_t nc = static_cast<uint8_t>(0.6*w + 3.2); return (0.6*w + 3.2) > nc ? nc + 1 : nc; }
```
or to make `return %s;` work, we need something like,
```c++
template<uint8_t w> constexpr auto nc125() noexcept {return (0.6*w + 3.2) > static_cast<uint8_t>(0.6*w + 3.2) ? static_cast<uint8_t>(0.6*w + 3.2) + 1 : static_cast<uint8_t>(0.6*w + 3.2); }
```
so that `get_nc_code = '(0.6*w + 3.2) > static_cast<uint8_t>(0.6*w + 3.2) ? static_cast<uint8_t>(0.6*w + 3.2) + 1 : static_cast<uint8_t>(0.6*w + 3.2)'` for `upsampfac==1.25`